### PR TITLE
Feature/python 2 to 3

### PR DESCRIPTION
**Pull request by jechols**
Migrates all the code to work with Python 3.  Tests pass, and clicking around the site works.  Will require updates to basically every plugin - I have these updates pending, but they should all merge at roughly the same time to minimize complications.

### CODE REVIEW

File: core/urls.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from django.conf import settings
 from django.utils import cache
 
-from views import home, browse, directory, reports, search, static
+from .views import home, browse, directory, reports, search, static
```
[the remark this answers is not in the document]
Comment (jduss4) [reply core/urls.py]: this question applies also to other files in this same commit

File: core/batch_loader.py [hunk 2]
```diff
@@ -2,8 +2,8 @@
 import os.path
 import re
 import logging
-import urllib2
-import urlparse
+import urllib.request, urllib.error, urllib.parse
+import urllib.parse
```
[the remark this answers is not in the document]
Comment (jechols) [reply core/batch_loader.py]: I just *really* love that package.

Actually it looks like a mistake the `2to3` converter made.  Lovely.  I'll fix it.

File: core/forms.py [hunk 3]
```diff
@@ -11,10 +11,10 @@
 
 # min and max year set in utils.py
 DAY_CHOICES = [(i, i) for i in range(1,32)]
-MONTH_CHOICES = ((1, u'Jan',), (2, u'Feb',), (3, u'Mar',),
-                 (4, u'Apr',), (5, u'May',), (6, u'Jun',),
-                 (7, u'Jul',), (8, u'Aug',), (9, u'Sep',),
-                 (10, u'Oct',), (11, u'Nov',), (12, u'Dec',))
+MONTH_CHOICES = ((1, 'Jan',), (2, 'Feb',), (3, 'Mar',),
```
[the remark this answers is not in the document]
Comment (jechols) [reply core/forms.py]: I take full credit for this fix that `2to3` did for me

File: core/management/commands/reconcile.py [hunk 4]
```diff
@@ -26,14 +26,14 @@ def handle(self, *args, **options):
...
         if len(missing_pages) > 0:
-            print "batches that are missing pages:"
+            print("batches that are missing pages:")
             for batch in missing_pages:
-                print "  %s has %s instead of %s pages" % (batch['name'],
-                        batch['my_page_count'], batch['page_count'])
-            print
+                print("  %s has %s instead of %s pages" % (batch['name'],
+                        batch['my_page_count'], batch['page_count']))
+            print()
```
[the remark this answers is not in the document]
Comment (jechols) [reply core/management/commands/reconcile.py]: The command as a whole, or just the extra `print()`?  The extra `print()` just spews an extra newline, but I know nothing of that command :(

File: core/urls.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jechols) [reply core/urls.py]: Python 3 thing - as far as I can tell you can't import from the current "package" without specifying a period before it.  If you're ever in doubt on this branch, look at commit fc95d68c469884b82f763af7cf2aa083bea224ec - that's the one that the automated tool did, so everything there is just what python 3 has to have.

File: core/title_loader.py [hunk 5]
```diff
@@ -54,7 +54,7 @@ def load_record(record):
 
             if self.records_processed % 1000 == 0:
                 _logger.info("processed %sk records in %.2f seconds" %
-                             (self.records_processed / 1000, seconds))
+                             (self.records_processed // 1000, seconds))
```
[the remark this answers is not in the document]
Comment (jechols) [reply core/title_loader.py]: I just made sure the thing did what it used to do... the whole change from Python 2 where "int in, int out" is no longer true means we have to be very careful how we divide anything.

File: core/management/commands/diff_batches.py [hunk 6]
```diff
@@ -26,7 +26,7 @@ def handle(self, batch_list_filename, *args, **options):
             raise CommandError('Usage is diff_batch %s' % self.args)
 
         batches = set()
-        batch_list = file(batch_list_filename)
+        batch_list = open(batch_list_filename)
```
[the remark this answers is not in the document]
Comment (jechols) [reply core/management/commands/diff_batches.py]: By default Python 3 opens files in "rt" mode, so I generally only had to change it when we needed the file to be opened in binary mode.  If you see anything that seems like it shouldn't be text mode, point it out, but I *think* we're good....

File: core/management/commands/reconcile.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jduss4) [reply core/management/commands/reconcile.py]: I was just curious why there was an extra newline, really, and if it was an accident of many years ago

[the remark this answers is not in the document]
Comment (jechols) [reply core/management/commands/reconcile.py]: Ah, gotcha - not sure what it's all about, maybe it just annoyed whoever built the thing to not have extra space :)

**Pull request merged**